user-service/controller/user-controller.js

```diff
@@ -19,3 +29,73 @@ export async function createUser(req, res) {
         return res.status(500).json({ message: 'Database failure when creating new user!' });
     }
 }
+export async function loginUser(req, res) {
+    // Check for existing username in db
+    const user = await getUser(req.body.username);
+
+    // if user does not exist
+    if (!user) {
+        return res.status(401).json({ message: 'Authentication failed. Invalid username.' });
+    }
+    const hashedPassword = user.password;

```

Comment:
Oh yes, I see the overlap here. Maybe I change my PR to return the user data instead? Then, can we merge my PR #34 first, before merging this? My implementation followed the original structure, for `UserModel` methods to be in `repository.js` instead of `user-orm.js`